src/main/java/frc/robot/auto/AutoRoutines.java

```diff
@@ -3,61 +3,194 @@
+				return null;
+		}
+	}
+
+	private Command checkElevatorCommands(AutoCommands commandEntry) {
+		switch (commandEntry) {
+			case ELEVATOR_GROUND_CMD:
+				return elevatorSystem.elevatorGroundCommand();
+			case ELEVATOR_L2_CMD:
+				return elevatorSystem.elevatorL2Command();
+			case ELEVATOR_L3_CMD:
+				return elevatorSystem.elevatorL3Command();
+			case ELEVATOR_L4_CMD:
+				return elevatorSystem.elevatorL4Command();
+			default:
+				return null;
+		}
+	}
+
+	private Command checkFunnelCommands(AutoCommands commandEntry) {
+		System.out.println("REACHED FUNNEL PARSE");
+		switch (commandEntry) {
+			case FUNNEL_OPEN_CMD:
+				return funnelSystem.openFunnelCommand();
+			case FUNNEL_CLOSE_CMD:
+				return funnelSystem.closeFunnelCommand();
+			default:
+				return null;
+		}
+	}
+
+	/* ----------------------------------------------------------- */
 
 	/**
-	 * Constructs an AutoRoutines object with the specified AutoFactory.
-	 * @param system1
+	 * Constructs an AutoRoutines object.
+	 * @param driveFSMSystem
+	 * @param elevatorFSMSystem
+	 * @param funnelFSMSystem
 	 * */
-	public AutoRoutines(DriveFSMSystem system1) {
-		driveSystem = system1;
+	public AutoRoutines(DriveFSMSystem driveFSMSystem, ElevatorFSMSystem elevatorFSMSystem,
+		FunnelFSMSystem funnelFSMSystem) {
 
-		autoFactory = driveSystem.createAutoFactory();
-		sysRoutine = autoFactory.newRoutine("AutoRoutine");
+		// Assign systems
+		driveSystem = driveFSMSystem;
+		elevatorSystem = elevatorFSMSystem;
+		funnelSystem = funnelFSMSystem;
 
-		paths = new HashMap<String, AutoTrajectory>();
-		commands = new HashMap<AutoCommands, Command>();
+		if (HardwareMap.isDriveHardwarePresent()) {
+			sysRoutine = driveSystem.createAutoFactory().newRoutine("AutoRoutine");
+			generateSysRoutineMap(Filesystem.getDeployDirectory().toString());
+		}
 
-		setupCommands();
-		generateSysRoutineMap(new File(Filesystem.getDeployDirectory(), "choreo"));
+		autoPaths = new AutoPaths();
 	}
 
 	/**
 	 * Creates and returns a auto routine that start with a path.
+	 * <br> <br>
+	 * For clarification with the throwException parameter, if you set it to FALSE,
+	 * it will just not add the command request into the returned SequentialCommandGroup.
+	 * This could be helpful for testing series of commands when you are working with partial
+	 * systems.
+	 * <br> <br>
+	 * If all systems are available, it should be set to TRUE - in an ideal
+	 * environment, no commands should throw this issue if all systems are available. It will be
+	 * more useful for debugging in this environment if you set throwException to true.
+	 *
 	 * @param autoStageSupply string of commands and trajectory names
+	 * @param throwException whether to throw an exception at a missing/unknown command or not.
 	 * @return the auto routine
 	 */
-	public AutoRoutine generateSequentialAutoWorkflow(Object[] autoStageSupply) {
-
+	public Command generateSequentialAutoWorkflow(Object[] autoStageSupply,
+		boolean throwException) {
```

Comment:
I don't think it's a good idea to be disabling exceptions as a parameter. I'd prefer you always throw the exception, and then try/catch them in the caller.